com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -595,9 +593,11 @@ private void NetworkVariableUpdate(ulong clientId, int behaviourIndex)
                     var shouldSend = false;
                     for (int k = 0; k < NetworkVariableFields.Count; k++)
                     {
-                        if (NetworkVariableFields[k].ShouldWrite(clientId, IsServer))
+                        var networkVariable = NetworkVariableFields[k];
+                        if (networkVariable.IsDirty() && networkVariable.CanClientRead(targetClientId))
```

Comment:
What is the rationale for copying the code from `ShouldWrite` and editing it here, as opposed to improving/adjusting/modifying `ShouldWrite`? Just wondering what will make the code more easily maintainable.